build breaks with libheif 1.20.1

### Describe the bug

libheif 1.21.0 seems to have removed some APIs that pillow_heif uses. 

### Steps/Code to Reproduce

Try building pillow_heif 1.0.0 with libheif 1.20.1 installed.

### Expected Results

Build succeeds.

### Actual Results

After the pkgsrc update for libheif to 1.21.0 I see:
```
gcc -O2 -g -fstack-clash-protection -I/usr/include -I/usr/pkg/include/python3.13 -I/usr/pkg/include -g -I/usr/include -I/usr/pkg/include/python3.13 -I/usr/pkg/include -fPIC -I/usr/include -I/usr/pkg/include/python3.13 -I
/usr/pkg/include -I/usr/pkg/include/python3.13 -c pillow_heif/_pillow_heif.c -o build/temp.netbsd-10.99.14-amd64-cpython-313/pillow_heif/_pillow_heif.o -O3 -Werror
pillow_heif/_pillow_heif.c: In function '_CtxImage_camera_intrinsic_matrix':
pillow_heif/_pillow_heif.c:1350:45: error: storage size of 'camera_intrinsic_matrix' isn't known
 1350 |         struct heif_camera_intrinsic_matrix camera_intrinsic_matrix;
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c:1352:14: error: implicit declaration of function 'heif_image_handle_has_camera_intrinsic_matrix' [-Werror=implicit-function-declaration]
 1352 |         if (!heif_image_handle_has_camera_intrinsic_matrix(self->handle)) {
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c:1355:25: error: implicit declaration of function 'heif_image_handle_get_camera_intrinsic_matrix' [-Werror=implicit-function-declaration]
 1355 |         if (check_error(heif_image_handle_get_camera_intrinsic_matrix(self->handle, &camera_intrinsic_matrix))) {
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c: In function '_CtxImage_camera_extrinsic_matrix_rot':
pillow_heif/_pillow_heif.c:1377:14: error: implicit declaration of function 'heif_image_handle_has_camera_extrinsic_matrix' [-Werror=implicit-function-declaration]
 1377 |         if (!heif_image_handle_has_camera_extrinsic_matrix(self->handle)) {
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c:1380:25: error: implicit declaration of function 'heif_image_handle_get_camera_extrinsic_matrix' [-Werror=implicit-function-declaration]
 1380 |         if (check_error(heif_image_handle_get_camera_extrinsic_matrix(self->handle, &camera_extrinsic_matrix))) {
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c:1380:25: error: incompatible type for argument 1 of 'check_error'
 1380 |         if (check_error(heif_image_handle_get_camera_extrinsic_matrix(self->handle, &camera_extrinsic_matrix))) {
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                         |
      |                         int
pillow_heif/_pillow_heif.c:14:35: note: expected 'struct heif_error' but argument is of type 'int'
   14 | int check_error(struct heif_error error) {
      |                 ~~~~~~~~~~~~~~~~~~^~~~~
pillow_heif/_pillow_heif.c:1383:17: error: implicit declaration of function 'heif_camera_extrinsic_matrix_get_rotation_matrix' [-Werror=implicit-function-declaration]
 1383 |         error = heif_camera_extrinsic_matrix_get_rotation_matrix(camera_extrinsic_matrix, rot);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pillow_heif/_pillow_heif.c:1383:17: error: incompatible types when assigning to type 'struct heif_error' from type 'int'
pillow_heif/_pillow_heif.c:1384:9: error: implicit declaration of function 'heif_camera_extrinsic_matrix_release' [-Werror=implicit-function-declaration]
 1384 |         heif_camera_extrinsic_matrix_release(camera_extrinsic_matrix);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```


### Versions

```shell
1.0.0
```

--- a/libheif/linux_build_libs.py
+++ b/libheif/linux_build_libs.py
@@ -13,7 +13,7 @@ PH_LIGHT_VERSION = sys.maxsize <= 2**32 or getenv("PH_LIGHT_ACTION", "0") != "0"
 
 LIBX265_URL = "https://bitbucket.org/multicoreware/x265_git/get/0b75c44c10e605fe9e9ebed58f04a46271131827.tar.gz"
 LIBDE265_URL = "https://github.com/strukturag/libde265/releases/download/v1.0.16/libde265-1.0.16.tar.gz"
-LIBHEIF_URL = "https://github.com/strukturag/libheif/releases/download/v1.19.8/libheif-1.19.8.tar.gz"
+LIBHEIF_URL = "https://github.com/strukturag/libheif/releases/download/v1.20.1/libheif-1.20.1.tar.gz"
 
 
 def download_file(url: str, out_path: str) -> bool:

--- a/libheif/macos/libheif.rb
+++ b/libheif/macos/libheif.rb
@@ -3,8 +3,8 @@
 class Libheif < Formula
   desc "ISO/IEC 23008-12:2017 HEIF file format decoder and encoder"
   homepage "https://www.libde265.org/"
-  url "https://github.com/strukturag/libheif/releases/download/v1.19.8/libheif-1.19.8.tar.gz"
-  sha256 "6c4a5b08e6eae66d199977468859dea3b5e059081db8928f7c7c16e53836c906"
+  url "https://github.com/strukturag/libheif/releases/download/v1.20.1/libheif-1.20.1.tar.gz"
+  sha256 "55cc76b77c533151fc78ba58ef5ad18562e84da403ed749c3ae017abaf1e2090"
   license "LGPL-3.0-only"
   # Set current revision from what it was taken plus 10
   revision 10

--- a/libheif/windows/mingw-w64-libheif/PKGBUILD
+++ b/libheif/windows/mingw-w64-libheif/PKGBUILD
@@ -4,7 +4,7 @@
 _realname=libheif
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-pkgver=1.19.8
+pkgver=1.20.1
 pkgrel=1
 pkgdesc="HEIF image decoder/encoder library and tools (mingw-w64)"
 arch=('any')
@@ -22,7 +22,7 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-libde265"
          "${MINGW_PACKAGE_PREFIX}-x265")
 source=("https://github.com/strukturag/libheif/releases/download/v${pkgver}/${_realname}-${pkgver}.tar.gz")
-sha256sums=('6c4a5b08e6eae66d199977468859dea3b5e059081db8928f7c7c16e53836c906')
+sha256sums=('55cc76b77c533151fc78ba58ef5ad18562e84da403ed749c3ae017abaf1e2090')
 
 build() {
   mkdir -p "${srcdir}"/build-${MSYSTEM} && cd "${srcdir}"/build-${MSYSTEM}

--- a/pi-heif/libheif/macos/libheif.rb
+++ b/pi-heif/libheif/macos/libheif.rb
@@ -3,8 +3,8 @@
 class Libheif < Formula
   desc "ISO/IEC 23008-12:2017 HEIF file format decoder and encoder"
   homepage "https://www.libde265.org/"
-  url "https://github.com/strukturag/libheif/releases/download/v1.19.8/libheif-1.19.8.tar.gz"
-  sha256 "6c4a5b08e6eae66d199977468859dea3b5e059081db8928f7c7c16e53836c906"
+  url "https://github.com/strukturag/libheif/releases/download/v1.20.1/libheif-1.20.1.tar.gz"
+  sha256 "55cc76b77c533151fc78ba58ef5ad18562e84da403ed749c3ae017abaf1e2090"
   license "LGPL-3.0-only"
   # Set current revision from what it was taken plus 10
   revision 10

--- a/pi-heif/libheif/windows/mingw-w64-libheif/PKGBUILD
+++ b/pi-heif/libheif/windows/mingw-w64-libheif/PKGBUILD
@@ -4,7 +4,7 @@
 _realname=libheif
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
-pkgver=1.19.8
+pkgver=1.20.1
 pkgrel=1
 pkgdesc="HEIF image decoder/encoder library and tools (mingw-w64)"
 arch=('any')
@@ -21,7 +21,7 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cmake"
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-libde265")
 source=("https://github.com/strukturag/libheif/releases/download/v${pkgver}/${_realname}-${pkgver}.tar.gz")
-sha256sums=('6c4a5b08e6eae66d199977468859dea3b5e059081db8928f7c7c16e53836c906')
+sha256sums=('55cc76b77c533151fc78ba58ef5ad18562e84da403ed749c3ae017abaf1e2090')
 
 build() {
   mkdir -p "${srcdir}"/build-${MSYSTEM} && cd "${srcdir}"/build-${MSYSTEM}

--- a/pillow_heif/_pillow_heif.c
+++ b/pillow_heif/_pillow_heif.c
@@ -2,6 +2,9 @@
 
 #include "Python.h"
 #include "libheif/heif.h"
+#if LIBHEIF_HAVE_VERSION(1,20,0)
+    #include "libheif/heif_properties.h"
+#endif
 #include "_ph_postprocess.h"
 
 /* =========== Common stuff ======== */

--- a/pillow_heif/_version.py
+++ b/pillow_heif/_version.py
@@ -1,3 +1,3 @@
 """Version of pillow_heif/pi_heif."""
 
-__version__ = "1.0.0"
+__version__ = "1.1.0.dev0"

--- a/tests/basic_test.py
+++ b/tests/basic_test.py
@@ -18,7 +18,7 @@ def test_libheif_info():
         assert key in info
 
     version = pillow_heif.libheif_version()
-    valid_prefixes = ["1.17.", "1.18.", "1.19."]
+    valid_prefixes = ["1.17.", "1.18.", "1.19.", "1.20."]
     assert any(version.startswith(prefix) for prefix in valid_prefixes)
 
 
@@ -107,7 +107,7 @@ def test_full_build():
     assert info["HEIF"]
     assert info["encoders"]
     assert info["decoders"]
-    expected_version = os.getenv("EXP_PH_LIBHEIF_VERSION", "1.19.8")
+    expected_version = os.getenv("EXP_PH_LIBHEIF_VERSION", "1.20.1")
     if expected_version:
         assert info["libheif"] == expected_version
 
@@ -118,7 +118,7 @@ def test_light_build():
     assert not info["AVIF"]
     assert not info["HEIF"]
     assert info["decoders"]
-    expected_version = os.getenv("EXP_PH_LIBHEIF_VERSION", "1.19.8")
+    expected_version = os.getenv("EXP_PH_LIBHEIF_VERSION", "1.20.1")
     if expected_version:
         assert info["libheif"] == expected_version
 

--- a/tests/read_test.py
+++ b/tests/read_test.py
@@ -533,14 +533,16 @@ def test_invalid_decoder():
         pillow_heif.options.PREFERRED_DECODER["HEIF"] = ""
 
 
-@pytest.mark.skipif(
-    parse_version(pillow_heif.libheif_version()) < parse_version("1.19.7"), reason="Requires libheif 1.19.7."
-)
-def test_200_megapixels():
-    with pytest.raises(RuntimeError):
-        _ = pillow_heif.open_heif("images/heif_special/200MP.heic").data
-    try:
-        pillow_heif.options.DISABLE_SECURITY_LIMITS = True
-        _ = pillow_heif.open_heif("images/heif_special/200MP.heic").data
-    finally:
-        pillow_heif.options.DISABLE_SECURITY_LIMITS = False
+# to-do: looks like we need now image with 400MP size to hit the security limits :(
+#
+# @pytest.mark.skipif(
+#     parse_version(pillow_heif.libheif_version()) < parse_version("1.19.7"), reason="Requires libheif 1.19.7"
+# )
+# def test_200_megapixels():
+#     with pytest.raises(RuntimeError):
+#         _ = pillow_heif.open_heif("images/heif_special/200MP.heic").data
+#     try:
+#         pillow_heif.options.DISABLE_SECURITY_LIMITS = True
+#         _ = pillow_heif.open_heif("images/heif_special/200MP.heic").data
+#     finally:
+#         pillow_heif.options.DISABLE_SECURITY_LIMITS = False

